### ISSUE TITLE
why use super small float -2^32+1 for padding

### ISSUE THREAD

**Issue by menglin0320** [Kyubyong/transformer#80]
As title

**Comment by Kyubyong** [comment]
Then after applying softmax, the value becomes zero.

**Comment by menglin0320** [comment]
I feel it's more intuitive to change softmax code

# softmax
        logits_mask = tf.sign(tf.abs(outputs))
        outputs = tf.exp(outputs)
        outputs = logits_mask * outputs
        epsilan = 1e-10
        outputs = outputs / (tf.reduce_sum(outputs, axis=(2), keepdims = True) + epsilan)
I modified this part
Also tf.layers.dense uses bias on default so Q, K, V does not perserve the zero paddings. I added mask for Q K and V.

**Comment by Kyubyong** [comment]
Oh, you're right. Thanks!
